### PR TITLE
Fix targeted refresh for nested stacks and their resource relations

### PR DESCRIPTION
**Pull request by andyvesel**
Authored by @Ladas from https://github.com/ManageIQ/manageiq-providers-openstack/pull/487

Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1728624

### CODE REVIEW

File: app/models/manageiq/providers/openstack/inventory/collector/target_collection.rb [hunk 1]
```diff
@@ -102,19 +103,31 @@ def orchestration_stacks
...
 
-  def get_orchestration_stack(stack_id, tenant_id)
-    tenant = memoized_get_tenant(tenant_id)
-    safe_get { @os_handle.detect_orchestration_service(tenant.try(:name)).stacks.get(stack_id) }
+  def indexed_all_orchestration_stacks
+    @indexed_all_orchestration_stacks ||= all_orchestration_stacks.index_by(&:id)
+  end
+
+  def get_orchestration_stack(stack_id, _tenant_id = nil)
+    # TODO fog needs to implement /v1/{tenant_id}/stacks/{stack_identity} call, right now the only supported call
```
Comment (Ladas) [inline app/models/manageiq/providers/openstack/inventory/collector/target_collection.rb]: didn't @aufi send PR for these? Or do I mistake it for something else?

**Comment by aufi** [comment]
LGTM based on previous comments, merging!

**Comment by simaishi** [comment]
@andyvesel @aufi All spec files conflit backporting to `hammer` branch. Can you please create a separate PR?

**Comment by andyvesel** [comment]
@simaishi sure, will do

**Comment by simaishi** [comment]
@andyvesel ping

**Comment by simaishi** [comment]
@andyvesel same thing on ivanchuk branch. Please create a separate PR.  If hammer/ivanchuk PRs will be identical, please just create a PR against ivanchuk and mark the PR as `hammer/yes`.

**Comment by simaishi** [comment]
@andyvesel  @aufi ping

**Comment by aufi** [comment]
@simaishi Hi, yes, I'm going to prepare backport PRs (today).

**Comment by aufi** [comment]
Having issues with infrastructure which postpones VCRs recording.

**Comment by simaishi** [comment]
Backported to hammer via https://github.com/ManageIQ/manageiq-providers-openstack/pull/550

**Comment by simaishi** [comment]
Backported to ivanchuk via https://github.com/ManageIQ/manageiq-providers-openstack/pull/551

**Pull request merged**